app/view_models/AtsForms.scala

```diff
@@ -17,15 +17,16 @@
 package view_models
 
 import com.google.inject.Inject
+import models.AtsYearChoice
 import play.api.data.Form
 import play.api.data.Forms._
 
 class AtsForms @Inject()() {
 
   val yearChoice = "year"
 
-  val atsYearFormMapping: Form[TaxYearEnd] = Form(
-    mapping(yearChoice -> optional(text).verifying("ats.select_tax_year.required", { _.isDefined }))(TaxYearEnd.apply)(
-      TaxYearEnd.unapply)
+  val atsYearFormMapping: Form[AtsYearChoice] = Form(
+    mapping(yearChoice -> (text).verifying("ats.select_tax_year.required", { !_.isEmpty }))(AtsYearChoice.fromString)(
```

Comment:
Prefer using `.nonEmpty` over `!.isEmpty` try to avoid negation as its easier to read